midend/inlining.cpp

```diff
@@ -356,6 +366,13 @@ const IR::Node* InlineDriver::preorder(IR::P4Program* program) {
         prog = prog->apply(*inliner);
         if (::errorCount() > 0)
             return prog;
+
+#if 0
+        // debugging code; we don't have an easy way to dump the program here,
+        // since we are not between passes
+        ToP4 top4(&std::cout, true, nullptr);
+        prog->apply(top4);
```

Comment:
It seems it would be useful to define an `opererator<<` and `ToP4::ToP4` constructor so you can do `LOG2(ToP4(prog))` or something similar...